libcontainer/cgroups/fs/cpuset.go

```diff
@@ -49,7 +52,112 @@ func (s *CpusetGroup) Remove(d *cgroupData) error {
 	return removePath(d.path("cpuset"))
 }
 
+func getCpusetStat(path string, filename string) ([]uint16, error) {
+	var extracted []uint16
+	fileContent, err := fscommon.GetCgroupParamString(path, filename)
+	if err != nil {
+		return extracted, err
+	}
+	if len(fileContent) == 0 {
+		return extracted, fmt.Errorf("%s found to be empty", filepath.Join(path, filename))
+	}
+
+	for _, s := range strings.Split(fileContent, ",") {
+		if !strings.Contains(s, "-") {
```

Comment:
I think it would be more optimal to do something like this

```go
splitted := strings.SplitN(s, "-", 3)
switch len(splitted) {
  case 3:
    return extracted, fmt.Errorf("invalid values in %s", filepath.Join(path, filename))
  case 2:
    // treat as min, max
  case 1:
    value, err := strconv.ParseUint(s, 10, 16)
}
```

This way you prevent the double-scanning of a string for `-`

Reply:
Applied